### DISCUSSION TITLE
Return a JSON response from page instead of a React component?

### DISCUSSION THREAD

**Discussion by msafi** [vercel/next.js#37937] [Help]
Next.js has very good caching and cache invalidation features in Static Site Generation and Incremental Static Regeneration. I want to use these features not to generate HTML, but to generate a JSON that my other HTML pages can use.

To generate this JSON, the route will initially request the data from my headless CMS. But, like regular React component pages, further requests will be cached and served for free from the cache. Then after the `revalidate` period has passed, or by using the `NextResponse.unstable_revalidate` feature, the JSON cache will be refreshed.

I don't think this feature is currently supported in Next.js, but when/if it comes, it would be very useful.

My question is, as a stopgap measure, is it safe to assume that `_next/data/<next.js build id>/[param].json` will always be there? Can I have my HTML pages request JSON data from routes under `_next/data` to achieve similar results to the process I described above?

Now if you're wondering, why not just build your JSON in the `getStaticProps` of the page itself? Why do you want the `getStaticProps` of your first page to request JSON data from your second page?

Well, I have 1000s of pages. They all need the same piece of data. If each one of them requests that piece of data from the headless CMS independently, I will have made 1000s requests to the headless CMS. If only one page makes that request and caches it, then my other pages can re-use the results of that request. In that case I would be making 1 request instead of 1000s. And the request/response latency would be a lot faster.

**Comment by msafi** [comment]
I found out that relying on the `_next/data/buildId` isn't feasible. The route doesn't work at all during development.

So I'm guessing this idea has to be properly built into Next.js to be usable.

**Comment by josealvarez97** [comment]
@msafi Did you ever find a way to accomplish this, at least with getStaticPaths (or getStaticProps)? I'd be okay not using incremental static regeneration.

**Reply by msafi** [reply]
Nope, no solution found, I'm afraid...
